src/main/java/shikabot/parser/Parser.java

```diff
@@ -0,0 +1,142 @@
+    }
+
+    private Command parseDoneCommand(String text) {
+        String str = text.substring(text.indexOf("done") + DONE_LENGTH).trim();
+        int index;
+        try {
+            index = Integer.parseInt(str) - 1;
+        } catch (NumberFormatException e) {
+            return new FailedCommand(NUMBER_FORMAT_ERROR);
+        }
+        if (index < 0) {
+            return new FailedCommand(NEGATIVE_INDEX_ERROR);
+        }
+        try {
+            return new DoneCommand(index);
+        } catch (IndexOutOfBoundsException e) {
+            return new FailedCommand(INVALID_TASK);
+        }
+    }
+
+    private Command parseDeleteCommand(String text) {
+        String str = text.substring(text.indexOf("delete") + DELETE_LENGTH).trim();
+        int index;
+        try {
+            index = Integer.parseInt(str) - 1;
+        } catch (NumberFormatException e) {
+            return new FailedCommand(NUMBER_FORMAT_ERROR);
+        }
+        if (index < 0) {
+            return new FailedCommand(NEGATIVE_INDEX_ERROR);
+        }
+        try {
+            return new DeleteCommand(index);
+        } catch (IndexOutOfBoundsException e) {
+            return new FailedCommand(INVALID_TASK);
+        }
+    }
+
+    private Command parseAddCommand(String text) {
+        Command command;
+        if (text.startsWith("todo")) {
+            try {
+                command = parseAddTodoCommand(text);
+            } catch (EmptyFieldException e) {
+                return new FailedCommand(EMPTY_FIELD);
+            }
+        } else if (text.startsWith("deadline")) {
+            try {
+                command = parseAddDeadlineCommand(text);
+            } catch (Task.InvalidTaskException e) {
+                return new FailedCommand(INVALID_DEADLINE_SYNTAX);
+            } catch (EmptyFieldException e) {
+                return new FailedCommand(EMPTY_FIELD);
+            }
+        } else {
+            try {
+                command = parseAddEventCommand(text);
+            } catch (Task.InvalidTaskException e) {
+                return new FailedCommand(INVALID_EVENT_SYNTAX);
+            } catch (EmptyFieldException e) {
+                return new FailedCommand(EMPTY_FIELD);
+            }
+        }
+        return command;
+    }
+
+    private Command parseAddTodoCommand(String text) throws EmptyFieldException {
+        String name = text.substring(text.indexOf("todo") + TODO_LENGTH).trim();
+        if (name.equals("")) throw new EmptyFieldException();
+        return new AddCommand('T', name, "");
+    }
+
+    private Command parseAddDeadlineCommand(String text) throws EmptyFieldException, Task.InvalidTaskException {
+        if (!text.contains("/by")) {
+            throw new Task.InvalidTaskException();
+        }
+        String name = text.substring(text.indexOf("deadline") + DEADLINE_LENGTH, text.indexOf("/")).trim();
+        String by = text.substring(text.indexOf("/by") + DIVIDER_LENGTH).trim();
```

Comment:
Perhaps strings in here can be in enums or constants variables to avoid the use of magic strings